mono/mini/method-to-ir.c

```diff
@@ -7304,8 +7304,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 	cfg->dont_inline = g_list_prepend (cfg->dont_inline, method);
 	if (cfg->method == method) {
 
-		if (cfg->prof_options & MONO_PROFILE_INS_COVERAGE)
-			cfg->coverage_info = mono_profiler_coverage_alloc (cfg->method, header->code_size);
+		cfg->coverage_info = mono_profiler_coverage_alloc (cfg->method, header->code_size);
```

Comment:
The installed filter functions will determine if anything actually gets allocated here. If no filter functions are allocated, `mono_profiler_coverage_alloc` returns `NULL`.